### PR TITLE
refactor(rules): defer book rules into a progressive disclosure skill

### PR DESCRIPTION
**Pull request by rjmurillo**
## Summary

Refs #3419

This PR moves the eight situational book-derived rules into one progressive disclosure skill. The everyday always-on synthesis stays in place. The instruction budget gate is ratcheted to the new lower code ceiling.

## Acceptance criteria

- Criterion 2: satisfied. Book rules now load through the task-triggered software engineering library skill.
- Budget ratchet: satisfied. Code extension ceilings now sit at 99,000 bytes.
- Criterion 5: satisfied. ADR-088 records the decision.
- Criterion 4: satisfied. The repo guide now points book depth to the on-demand skill.
- Criterion 3: not included. Full metadata normalization stays a separate follow-up.

## Budget evidence

| Ext | Old bytes | New bytes | New ceiling | Status |
| --- | ---: | ---: | ---: | --- |
| .cs | 218480 | 94641 | 99000 | PASS |
| .md | 81622 | 81622 | 83000 | PASS |
| .ps1 | 218312 | 94473 | 99000 | PASS |
| .py | 218603 | 94764 | 99000 | PASS |

Actual gate output:

```text
Always-On Instruction Budget (language baseline)

Ext     Files     Bytes   Ceiling   Tokens~    Usage  Status
------------------------------------------------------------
.cs        11     94641     99000     25041    95.6%    PASS
.md         9     81622     83000     21564    98.3%    PASS
.ps1       11     94473     99000     25074    95.4%    PASS
.py        11     94764     99000     25137    95.7%    PASS

PASS: All languages within the always-on instruction ceiling.
```

## Validation

- `python3 -m scripts.validation.instruction_budget`: PASS.
- `uv run --frozen python -m pytest tests/ -x -q -k "instruction_budget or generate_rules or drift"`: 269 passed, 15140 deselected.
- `uv run --frozen python -m pytest tests/test_validate_workspace_budget.py -q`: 19 passed.
- Pre-push hook: 15382 passed, 27 skipped, 2 warnings.
- Pre-PR validation: 31 passed, 4 skipped, 0 failed.
- Drift, plugin parity, plugin version bump, and task-surface orphan scan passed.

## Version

Plugin version: 0.6.141.

## References

ADR:

- `.agents/architecture/ADR-088-progressive-disclosure-book-rules.md`

Created:

- `.claude/skills/software-engineering-library/SKILL.md`
- `.claude/skills/software-engineering-library/references/clean-architecture.md`
- `.claude/skills/software-engineering-library/references/data-intensive-applications.md`
- `.claude/skills/software-engineering-library/references/domain-driven-design.md`
- `.claude/skills/software-engineering-library/references/enterprise-patterns.md`
- `.claude/skills/software-engineering-library/references/philosophy-of-software-design.md`
- `.claude/skills/software-engineering-library/references/refactoring.md`
- `.claude/skills/software-engineering-library/references/release-it.md`
- `.claude/skills/software-engineering-library/references/working-with-legacy-code.md`
- `src/copilot-cli/skills/software-engineering-library/SKILL.md`
- `src/copilot-cli/skills/software-engineering-library/references/clean-architecture.md`
- `src/copilot-cli/skills/software-engineering-library/references/data-intensive-applications.md`
- `src/copilot-cli/skills/software-engineering-library/references/domain-driven-design.md`
- `src/copilot-cli/skills/software-engineering-library/references/enterprise-patterns.md`
- `src/copilot-cli/skills/software-engineering-library/references/philosophy-of-software-design.md`
- `src/copilot-cli/skills/software-engineering-library/references/refactoring.md`
- `src/copilot-cli/skills/software-engineering-library/references/release-it.md`
- `src/copilot-cli/skills/software-engineering-library/references/working-with-legacy-code.md`
- `.agents/analysis/ADR-088-debate-log.md`
- `.agents/critique/ADR-088-debate-log.md`
- `.agents/sessions/2026-07-27-session-3419-progressive-disclosure.json`

Deleted source rules:

- `.claude/rules/clean-architecture.md`
- `.claude/rules/data-intensive-applications.md`
- `.claude/rules/domain-driven-design.md`
- `.claude/rules/enterprise-patterns.md`
- `.claude/rules/philosophy-of-software-design.md`
- `.claude/rules/refactoring.md`
- `.claude/rules/release-it.md`
- `.claude/rules/working-with-legacy-code.md`

Deleted generated instruction mirrors:

- `.github/instructions/clean-architecture.instructions.md`
- `.github/instructions/data-intensive-applications.instructions.md`
- `.github/instructions/domain-driven-design.instructions.md`
- `.github/instructions/enterprise-patterns.instructions.md`
- `.github/instructions/philosophy-of-software-design.instructions.md`
- `.github/instructions/refactoring.instructions.md`
- `.github/instructions/release-it.instructions.md`
- `.github/instructions/working-with-legacy-code.instructions.md`
- `src/copilot-cli/instructions/clean-architecture.instructions.md`
- `src/copilot-cli/instructions/data-intensive-applications.instructions.md`
- `src/copilot-cli/instructions/domain-driven-design.instructions.md`
- `src/copilot-cli/instructions/enterprise-patterns.instructions.md`
- `src/copilot-cli/instructions/philosophy-of-software-design.instructions.md`
- `src/copilot-cli/instructions/refactoring.instructions.md`
- `src/copilot-cli/instructions/release-it.instructions.md`
- `src/copilot-cli/instructions/working-with-legacy-code.instructions.md`

### CODE REVIEW

**Comment by rjmurillo** [comment]
Review follow-up pushed at f55ed31126928d52755009bd9ef8f241bc3cce92.

## Reachability fixes

`git grep -n software-engineering-library HEAD -- .claude/commands .claude/skills/analyze .claude/skills/autoplan .claude/agents` now returns:

```text
HEAD:.claude/commands/spec.md:408:   Elicit answers to the seven ontology prompts, in order. Each is grounded in the domain-driven-design reference in `software-engineering-library`:
HEAD:.claude/skills/analyze/SKILL.md:119:When analysis discovers book-depth conditions, invoke `software-engineering-library` and open the matching reference before recommending the fix. Conditions include low test coverage, old file age, hard-to-test seams, external API calls, queues, retries, transaction boundaries, event ordering, schema evolution, layer dependency direction, bounded context splits, and module interface shape. This keeps the heavy guidance deferred until evidence shows it matters.
HEAD:.claude/skills/autoplan/SKILL.md:100:| Software design depth, architecture boundaries, domain modeling, refactoring, legacy code, low coverage, old files, external APIs, queues, retries, transactions, event ordering, schema evolution, resilience | Skill: software-engineering-library, then the routed reference |
```

## Activation tests

All eight passed:

- `test_activation_fixture_routes_clean_architecture_to_library`
- `test_activation_fixture_routes_domain_driven_design_to_library`
- `test_activation_fixture_routes_enterprise_patterns_to_library`
- `test_activation_fixture_routes_refactoring_to_library`
- `test_activation_fixture_routes_working_with_legacy_code_to_library`
- `test_activation_fixture_routes_data_intensive_applications_to_library`
- `test_activation_fixture_routes_release_it_to_library`
- `test_activation_fixture_routes_philosophy_of_software_design_to_library`

Command result: 8 passed, 33 deselected.

## Eval evidence

`uv run --frozen python scripts/eval/eval-rule-activation.py --scenarios tests/evals/rule-scenarios/*.json --dry-run` now succeeds.

Dry-run activation plan:

- clean-architecture: 3 scenarios, 18 calls, description present.
- data-intensive-applications: 3 scenarios, 18 calls, description present.
- domain-driven-design: 3 scenarios, 18 calls, description present.
- enterprise-patterns: 3 scenarios, 18 calls, description present.
- philosophy-of-software-design: 3 scenarios, 18 calls, description present.
- refactoring: 4 scenarios, 24 calls, description present.
- release-it: 3 scenarios, 18 calls, description present.
- unified-software-engineering: 4 scenarios, 24 calls, description present.
- working-with-legacy-code: 4 scenarios, 24 calls, description present.

Total planned calls: 180. Estimated tokens: about 630,000. Live scoring was not run because this environment has no `ANTHROPIC_API_KEY`.

## Rollback trigger

ADR-088 now says to revert this move if the activation eval reports fewer than 7 of 8 moved references passing in two consecutive runs, or if any one moved reference has no positive activation signal in a run.

## Orphan validator issue

Filed #3554 for deleted rule path references not being detected by orphan-ref-validator. It cross-references #3456 as a different scanner gap.

## Reference triage

- Live references: fixed. Current live deleted-rule ref count is 0.
- Historical records: left intact. Current historical deleted-rule ref count is 61.
- ADRs: left as historical decision records except ADR-088, which was amended because it is the active decision for this PR.
- Current rules: fixed canonical `unified-software-engineering.md` and regenerated both instruction mirrors.

## Validation

- Targeted eval and reference tests: 574 passed.
- Eight activation routing tests: 8 passed, 33 deselected.
- Original targeted suite: 269 passed, 15150 deselected.
- Pre-PR validation: 31 passed, 4 skipped, 0 failed.
- Instruction budget: PASS, .py 94741, .cs 94618, .ps1 94450, .md 81622.
- Drift, plugin parity, and plugin version bump gates: PASS.
- Pre-push completed with exit 0.

**Comment by rjmurillo** [comment]
PR #3550 has pushed the general orphan-ref validator capability for deleted rule paths. It adds rule_path and instruction_path checks for backticked paths and Markdown link targets, so this branch should see deleted rule references once #3550 lands or is merged into the branch.

Follow-up #3556 tracks JSON, YAML, frontmatter, bare fenced-code, and Python-literal extraction.

**Comment by rjmurillo** [comment]
owner: local:pr-autofix
session: session-3507-autofix
acquired_at: 2026-07-28T03:34:43Z
expires_at: 2026-07-28T03:49:43Z
base_sha: 9c49ac4996f3b77b7827544b9867f45b872bc057

File: src/copilot-cli/instructions/generated-artifacts.instructions.md [hunk 1]
```diff
@@ -169,7 +169,7 @@ never have to uninstall to recover from an artifact we generated.
 - `.claude/skills/agent-harness-reference/references/official-hook-contracts.md`. Official sources and refresh procedure.
 - `.claude/skills/ai-agents-portability-campaign/SKILL.md`. Contract change workflow.
 - `.claude/rules/canonical-source-mirror.md`. Self-referential test anti-pattern.
-- `.claude/rules/release-it.md`. Fail fast and loud; bound the blast radius by prevention, not by silently swallowing failures.
+- `.claude/skills/software-engineering-library/references/release-it.md`. Fail fast and loud; bound the blast radius by prevention, not by silently swallowing failures.
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply src/copilot-cli/instructions/generated-artifacts.instructions.md:172]: This is a pre-existing limitation of the mirror generation pipeline (`build/scripts/generate_rules.py`). It copies `.claude/rules/*.md` content verbatim into `src/copilot-cli/instructions/` without rewriting internal `.claude/` paths to their in-plugin equivalents. The canonical source (`.claude/rules/generated-artifacts.md`) uses `.claude/` paths which are correct in that context. Path rewriting in the generation step is a separate enhancement (will file follow-up).

File: src/copilot-cli/skills/requirements-interview/SKILL.md [hunk 2]
```diff
@@ -112,6 +112,6 @@ Every PRD section reaches the spec-generator unchanged so it does not re-ask que
 - Upstream: <https://github.com/mattpocock/skills>
 - [Circle of Competence](references/mental-models-circle-of-competence.md) - Calibrate confidence in a recommended answer by whether the decision sits inside the team's tested knowledge
 - Related skills: `decision-critic`, `pre-mortem`, `cynefin-classifier`
-- Related rules: `.claude/rules/clean-architecture.md`, `.claude/rules/domain-driven-design.md`, `.claude/rules/security.md`
+- Related references: `software-engineering-library` clean-architecture and domain-driven-design references, plus `.claude/rules/security.md`
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply src/copilot-cli/skills/requirements-interview/SKILL.md:115]: Same mirror generation limitation as thread above. The canonical source (`.claude/skills/requirements-interview/SKILL.md`) uses `.claude/rules/security.md` correctly in that context. The skill generation pipeline (`build/scripts/generate_skills.py`) copies verbatim without path rewriting. Will file a single follow-up issue for both path-rewriting gaps.

File: .agents/sessions/2026-07-27-session-3419-progressive-disclosure.json [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    ],
+    "decisions": [
+      {
+        "decision": "Keep code-quality, pragmatic-programmer, and unified-software-engineering always-on while moving only the eight situational book rules.",
+        "rationale": "Everyday engineering rules need passive adherence, while book-depth material should load by task.",
+        "impact": "Code edits load about 95 KB of instructions instead of about 218 KB, while book-depth guidance remains available on task trigger."
+      }
+    ]
+  },
+  "endingCommit": "",
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .agents/sessions/2026-07-27-session-3419-progressive-disclosure.json]: Fixed in 6ac7be5ba. Populated `endingCommit` with `f55ed31126928d52755009bd9ef8f241bc3cce92` (the last non-merge commit on the branch before the PR was opened).

File: .agents/architecture/ADR-088-progressive-disclosure-book-rules.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
+---
+id: ADR-088
+status: proposed
+date: 2026-07-27
+decision-makers: []
+supersedes: []
+superseded-by: null
+explainer: null
+implemented: true
+---
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .agents/architecture/ADR-088-progressive-disclosure-book-rules.md:10]: Fixed in 6ac7be5ba. Set `implemented: false` in ADR-088 frontmatter. It will flip to `true` post-merge per ADR-073 lifecycle convention.

File: scripts/eval/eval-rule-activation.py [hunk 5]
```diff
@@ -540,12 +563,79 @@ def _resolve_rule_path(rule_path_str: str) -> Path | int:
...
+            file=sys.stderr,
+        )
+        return 2
+    if not isinstance(reference_path_str, str) or not reference_path_str.strip():
+        print(
+            f"ERROR: reference_path must be a non-empty string in {spath}",
+            file=sys.stderr,
+        )
+        return 2
+    return _resolve_skill_reference(skill_path_str.strip(), reference_path_str.strip())
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply scripts/eval/eval-rule-activation.py:803]: Fixed in 6ac7be5ba. Added a fail-fast guard at the top of `_resolve_target_paths()`: if both `rule_path` and `skill_path` are non-empty strings, it prints an error and returns exit code 2. This prevents silent evaluation of the wrong target when the config is ambiguous.

File: tests/eval/test_optimize_artifact_cli.py [hunk 6]
```diff
@@ -405,7 +405,7 @@ class TestExtractRealRuleEnvelope:
     """The shape eval-rule-activation.py --output actually writes.
 
     Verified against a live run over tests/evals/rule-scenarios/*.json: the
-    file is {"rules": {<rule-name>: {"rule_path", "scenarios", "summary"}}}.
+    file is {"rules": {<rule-name>: {"target_path", "scenarios", "summary"}}}.
     Scenario ids restart at S1 inside every rule, so 24 real scenarios carry
     only 4 distinct ids and must be namespaced before they can be task ids.
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply tests/eval/test_optimize_artifact_cli.py:569]: Fixed in 6ac7be5ba. Updated the docstring to document `reference_path` in the output shape: `{"rules": {<rule-name>: {"target_path", "reference_path", "scenarios", "summary"}}}`. Added note that `reference_path` is present for progressive-disclosure skill reference targets.

File: scripts/eval/README.md [hunk 7]
```diff
@@ -153,11 +155,12 @@ Per-rule scenario files live in `tests/evals/rule-scenarios/{rule}.json`:
...
-Adding a new rule eval:
+Adding a new activation eval:
 
 1. Write `tests/evals/rule-scenarios/{rule-id}.json` with 3-5 positive scenarios and at least one negative case.
-2. Run `python3 scripts/eval/eval-rule-activation.py --scenarios tests/evals/rule-scenarios/{rule-id}.json --dry-run` to confirm the script can parse the rule.
-3. Run live (without `--dry-run`) to score. Cost is ~$0.25 per rule (24 calls × ~3500 tokens).
+2. Use `rule_path` for always-on rules, or `skill_path` plus `reference_path` for progressive-disclosure references.
+3. Run `uv run python scripts/eval/eval-rule-activation.py --scenarios tests/evals/rule-scenarios/{rule-id}.json --dry-run` to confirm the script can parse the target.
+4. Run live (without `--dry-run`) to score. Cost is ~$0.25 per target (24 calls × ~3500 tokens).
 4. Iterate on the rule's `description` field until the `description` mechanism scores within 0.5 of `full`. That is the signal the rule is activatable from frontmatter alone.
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply scripts/eval/README.md]: Fixed in 6ac7be5ba (then resolved during merge with updated content from the remote branch). The numbered list now reads 4. / 5. with updated wording from a parallel commit.

File: src/copilot-cli/skills/spec/SKILL.md [hunk 8]
```diff
@@ -3,6 +3,7 @@ name: spec
 description: Define what to build. Transform a problem into testable requirements with acceptance criteria.
 argument-hint: problem-statement-or-issue-number
 allowed-tools: Task, Skill, Read, Write, Glob, Grep
+size-exception: true
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply src/copilot-cli/skills/spec/SKILL.md:6]: The `size-exception: true` frontmatter was not introduced by this PR (it pre-dates it). The `/spec` skill is a multi-phase workflow prompt (627 lines) that cannot be mechanically split without breaking the workflow. Justification: the prompt encodes a sequential interview-to-specification pipeline where earlier phases reference later validation criteria. Adding inline justification is a reasonable follow-up but out of scope for this PR.

File: .claude/commands/spec.md [hunk 9]
```diff
@@ -2,6 +2,7 @@
 description: Define what to build. Transform a problem into testable requirements with acceptance criteria.
 allowed-tools: Task, Skill, Read, Write, Glob, Grep
 argument-hint: problem-statement-or-issue-number
+size-exception: true
 ---
 
 @CLAUDE.md
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .claude/commands/spec.md:8]: Same as above: `size-exception: true` in `.claude/commands/spec.md` pre-dates this PR. The command mirrors the copilot-cli skill. Out of scope for this change.

File: .agents/memory/episodes/episode-2026-07-27-session-3419-progressive-disclosure.json [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+      "caused_by": [
+        "e014"
+      ],
+      "leads_to": []
+    },
+    {
+      "id": "e009",
+      "timestamp": "2026-07-27T00:00:00+00:00",
+      "type": "commit",
+      "content": "Commit: f55ed31126928d52755009bd9ef8f241bc3cce92",
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .agents/memory/episodes/episode-2026-07-27-session-3419-progressive-disclosure.json:191]: Fixed in 1ce0f3e3b. Renamed the duplicate `e009` (commit type) to `e016`. The collision occurred when a merge milestone and a commit record were both assigned sequential id `e009`.

**Comment by rjmurillo** [comment]
owner: none
session: session-3507-autofix
acquired_at: 2026-07-28T07:15:10Z
expires_at: 2026-07-28T07:15:10Z
base_sha: 0000000000000000000000000000000000000000
